### PR TITLE
Add benchmark support for e2e tests

### PR DESCRIPTION
**Pull request by erman-gurses**

### CODE REVIEW

File: shark_turbine/kernel/wave/utils.py [hunk 1]
```diff
@@ -366,7 +369,29 @@ def compile_and_invoke(
         _invoke(ctx.vm_context, device, func, kernel_inputs, kernel_outputs)
 
     if run_bench:
-        inputs = [inp.numpy() for inp in kernel_inputs]
+        bench_with_constant_weights = config.get("backend", False)
+        tempfiles = []
+        inputs = []
+        if bench_with_constant_weights:
```
Comment (Hardcode84) [inline shark_turbine/kernel/wave/utils.py:375]: Why do we need this?

File: shark_turbine/kernel/wave/utils.py [hunk 2]
```diff
@@ -366,7 +369,29 @@ def compile_and_invoke(
...
+            for inp in kernel_inputs:
+                inputs.append(
+                    "x".join(
+                        [str(x) for x in inp.shape]
+                        + [TORCH_DTYPE_TO_SIGNED_MLIR_TYPE_ASM[inp.dtype]]
+                    )
+                )
+        else:
+            for inp in kernel_inputs:
+                tf = tempfile.NamedTemporaryFile(delete=False)
```
Comment (Hardcode84) [inline shark_turbine/kernel/wave/utils.py:385]: Why `delete=False`?

File: shark_turbine/kernel/wave/utils.py [hunk 3]
```diff
@@ -375,6 +400,8 @@ def compile_and_invoke(
             **benchmark_flags,
         )
         _print_bench_result(benchmark_results, bench_file)
+        for tf in tempfiles:
```
Comment (Hardcode84) [inline shark_turbine/kernel/wave/utils.py:403]: The files will closed when `tempfiles` list is destroyed (or or can `del tempfiles` manually), don't need to call `close` on each file manually.

File: shark_turbine/kernel/wave/utils.py [hunk 4]
```diff
@@ -366,7 +369,29 @@ def compile_and_invoke(
         _invoke(ctx.vm_context, device, func, kernel_inputs, kernel_outputs)
 
     if run_bench:
-        inputs = [inp.numpy() for inp in kernel_inputs]
+        bench_with_constant_weights = config.get("backend", False)
```
Comment (Hardcode84) [inline shark_turbine/kernel/wave/utils.py:372]: Wrong config option name.

File: tests/kernel/wave/wave_e2e_test.py [hunk 5]
```diff
@@ -101,6 +101,7 @@ def test(
         },
         canonicalize=True,
         run=True,
+        run_bench=True,
```
Comment (Hardcode84) [inline tests/kernel/wave/wave_e2e_test.py:104]: This should not be enabled by default, we need a separate option. Also, where bench results are saved?

File: shark_turbine/kernel/wave/utils.py (hunk 3 above)
Comment (erman-gurses) [reply shark_turbine/kernel/wave/utils.py:403]: Do you mean when we have "delete=True", we do not need to close the file?

File: shark_turbine/kernel/wave/utils.py (hunk 4 above)
Comment (erman-gurses) [reply shark_turbine/kernel/wave/utils.py:372]: Can you elaborate?

**Review by Hardcode84 — APPROVED** [review]

**Pull request merged**